File: app/src/main/java/com/commit451/gitlab/api/rss/UriTransform.java
package com.commit451.gitlab.api.rss;

import android.net.Uri;

import com.commit451.gitlab.model.Account;
import com.commit451.gitlab.util.ConversionUtil;

import org.simpleframework.xml.transform.Transform;

/**
 * Uri Transformer
 */
public class UriTransform implements Transform<Uri> {
    private final Account account;

    public UriTransform(Account account) {
        this.account = account;
    }

    @Override
    public Uri read(String value) throws Exception {
        return ConversionUtil.toUri(account, value);
    }

    @Override
    public String write(Uri value) throws Exception {
        return ConversionUtil.fromUri(value);
    }
}

File: app/src/main/java/com/commit451/gitlab/api/rss/UriTransform.kt
package com.commit451.gitlab.api.rss

import android.net.Uri

import com.commit451.gitlab.model.Account
import com.commit451.gitlab.util.ConversionUtil

import org.simpleframework.xml.transform.Transform

/**
 * Uri Transformer
 */
class UriTransform(private val account: Account) : Transform<Uri> {

    @Throws(Exception::class)
    override fun read(value: String): Uri? {
        return ConversionUtil.toUri(account, value)
    }

    @Throws(Exception::class)
    override fun write(value: Uri): String? {
        return ConversionUtil.fromUri(value)
    }
}
